src/SonarQube.Client.Tests/Api/SonarQubeService_GetAllOrganizationsAsync.cs

```diff
@@ -0,0 +1,192 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.Tests.Api
+{
+    [TestClass]
+    public class SonarQubeService_GetAllOrganizationsAsync : SonarQubeService_TestBase
+    {
+        [TestMethod]
+        public async Task GetOrganizations_ExampleFromSonarQube()
+        {
+            await ConnectToSonarQube("6.2.0.0");
+
+            SetupRequest("api/organizations/search?p=1&ps=500",
+                @"
+{
+  ""paging"": {
+    ""pageIndex"": 1,
+    ""pageSize"": 25,
+    ""total"": 2
+  },
+  ""organizations"": [
+    {
+      ""key"": ""foo-company"",
+      ""name"": ""Foo Company"",
+      ""guarded"": true
+    },
+    {
+      ""key"": ""bar-company"",
+      ""name"": ""Bar Company"",
+      ""description"": ""The Bar company produces quality software too."",
+      ""url"": ""https://www.bar.com"",
+      ""avatar"": ""https://www.bar.com/logo.png"",
+      ""guarded"": false
+    }
+  ]
+}");
+
+            var result = await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(2);
+            result.Select(x => x.Key).Should().BeEquivalentTo(new[] { "foo-company", "bar-company" });
+            result.Select(x => x.Name).Should().BeEquivalentTo(new[] { "Foo Company", "Bar Company" });
+        }
+
+        [TestMethod]
+        public async Task GetOrganizations_Paging()
+        {
+            await ConnectToSonarQube("6.2.0.0");
+
+            SetupRequest("api/organizations/search?p=1&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 1,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(1, 500).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            SetupRequest("api/organizations/search?p=2&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 2,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(501, 500).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            SetupRequest("api/organizations/search?p=3&ps=500", $@"
+{{
+  ""paging"": {{
+    ""pageIndex"": 3,
+    ""pageSize"": 500,
+    ""total"": 2
+  }},
+  ""organizations"": [
+    {string.Join(",\n", Enumerable.Range(1001, 10).Select(i => $@"{{ ""key"": ""{i}"", ""name"": ""Name{i}"" }}"))}
+  ]
+}}");
+
+            var result = await service.GetAllOrganizationsAsync(CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(1010);
```

Comment:
Fine by me but why did you chose 1010?

Reply:
It could be 1001... First page has 500 items, second - 500, third 10 => 1010.

Reply:
Also it is a nice number - if `1010` is binary, it converts to `10` in decimal 💃 🔢